.buildkite/scripts/validate-tag.sh

```diff
@@ -0,0 +1,23 @@
+#!/bin/bash
+set -euo pipefail
+
+source .buildkite/scripts/tooling.sh
+
+transformTagAndValidate() {
+    local version=$1
+    version=${version/#v/}
+
+    if [[ $version =~ ^[0-9]+.[0-9]+.[0-9]+(-[A-Za-z0-9_]+)?$ ]]; then
+        echo "valid version: ${version}"
+        DOCKER_TAG=${version}
+    else
+        echo "invalid version: ${version}"
+        echo "unsupported docker tag, please use the major.minor.path(-prerelease)? format (for example: 1.2.3 or 1.2.3-alpha)."
+        exit 1
+    fi
+}
+
+DOCKER_TAG=$(buildkite-agent meta-data get DOCKER_TAG)
```

Comment:
I didn't test caller-pipeline behavior... need to merge and look from another repo (test from fleet). By default, it is "latest" as in Jenkins, and if it is called from UI and the pipeline user erases the pre-filled value `latest` it will not go further because the field is required

Reply:
That default value "latest" would not be valid in Jenkins since it will not match with the regex:
https://github.com/elastic/package-registry/blob/main/.ci/release-package-registry-distribution.groovy#L77

For now, I would not allow triggering this pipeline and pushing images with latest. This job is not meant to push these docker distribution images with docker tag `latest`.